compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3324,6 +3377,19 @@ public Boolean visit(BTupleType target, BType source) {
                     sourceRestType);
         }
 
+        private boolean hasCommonOrderedTypeForTuples(BType baseType, List<BType> typeList, int startIndex) {
```

Comment:
Since the `baseType = typeList.get(startIndex - 1)`, can we create it here, without passing as a param?